src/gsMatrix/gsSparseMatrix.h

```diff
@@ -433,6 +433,40 @@ class gsSparseMatrix : public Eigen::SparseMatrix<T,_Options,_Index>
         return result;
     }
 
+    index_t bandWidth(index_t upto = std::numeric_limits<index_t>::max())  const
```

Comment:
bandwidth is something else:
https://en.wikipedia.org/wiki/Band_matrix

Better: maxNonzerosPerCol

Fixing per row or per col is either inefficient for RowMajor or ColMajor

Reply:
fixed, the allocation is accordingly..